### PR TITLE
Bug fix, performance improvements (quri-parts-itensor)

### PR DESCRIPTION
**Pull request by toru4838**
Bug fix and performance improvements:

- Add a flag to ensure_itensor_loaded()
- Normalize MPS before executing estimator/sampler
- Add license terms
- Fix typos

### CODE REVIEW

File: packages/itensor/quri_parts/itensor/estimator.py [hunk 1]
```diff
@@ -57,15 +66,19 @@ def _estimate(
...
-    exp: float = jl.expectation(psi, op)
 
     # See https://github.com/QunaSys/quri-parts/pull/203#discussion_r1329458816
     error = 0.0
     if "maxdim" in kwargs or "cutoff" in kwargs:
         error = np.nan
 
+    # calculate expectation value
+    if kwargs:
+        psi = jl.normalize(psi)
```
Comment (kwkbtr) [inline packages/itensor/quri_parts/itensor/estimator.py]: What's the rationale of this `if kwargs:` condition?

Comment (kwkbtr) [reply packages/itensor/quri_parts/itensor/estimator.py]: In my understanding, the state should always be normalized.
I guess it is the case that the state can be unnormalized when some options such as `maxdim` or `cutoff` are specified.
With that assumption:

* If calling `.normalize()` doesn't have performance penalty when the state is already normalized, then we can always call it without the `if` condition.
* If it is not the case, then I think the options (`maxdim`, `cutoff`, etc.) should explicitly be specified in the `if` condition.

Comment (toru4838) [reply packages/itensor/quri_parts/itensor/estimator.py]: Thanks for the comments!

> In my understanding, the state should always be normalized.
> I guess it is the case that the state can be unnormalized when some options such as maxdim or cutoff are specified.

That's true.

> If calling .normalize() doesn't have performance penalty when the state is already normalized, then we can always call it without the if condition.

Every time you call `normalize()`, the norm (lognorm actually) of the MPS is computed regardless of its norm. So avoiding the call of `normalize()` when the MPS is normalized can be better.
https://github.com/ITensor/ITensors.jl/blob/e7740ebb8df26cbacc1671113656e702030e43bb/src/mps/abstractmps.jl#L1299-L1321

> If it is not the case, then I think the options (maxdim, cutoff, etc.) should explicitly be specified in the if condition.

I thought there might be an option which changes the norm of the MPS which I don't recognize. I just made a low risk choice (I should say I was a bit lazy 😅). I will change it to check if `maxdim`, `cutoff`, or `mindim` is specified.

**Review by kwkbtr — APPROVED** [review]
👍

**Pull request merged**